pkg/adsc/adsc.go

```diff
@@ -309,44 +309,11 @@ func getPrivateIPIfAvailable() net.IP {
 }
 
 func (a *ADSC) tlsConfig() (*tls.Config, error) {
-	var clientCert tls.Certificate
 	var clientCerts []tls.Certificate
 	var serverCABytes []byte
 	var err error
-	var certName string
 
-	var getClientCertificate func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error)
-
-	if a.cfg.SecretManager != nil {
-		getClientCertificate = func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			tok, err := ioutil.ReadFile(a.cfg.JWTPath)
-			if err != nil {
-				log.Infof("Failed to get credential token in agent: %v", err)
-				tok = []byte("")
-			}
-
-			certName = fmt.Sprintf("(generated from %s)", a.cfg.JWTPath)
-			key, err := a.cfg.SecretManager.GenerateSecret(context.Background(), "agent",
-				cache.WorkloadKeyCertResourceName, string(tok))
-			if err != nil {
-				return nil, err
-			}
-			clientCert, err = tls.X509KeyPair(key.CertificateChain, key.PrivateKey)
-			if err != nil {
-				return nil, err
-			}
-			return &clientCert, nil
-		}
-	} else if a.cfg.CertDir != "" {
-		getClientCertificate = func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			certName = a.cfg.CertDir + "/cert-chain.pem"
-			clientCert, err = tls.LoadX509KeyPair(certName, a.cfg.CertDir+"/key.pem")
-			if err != nil {
-				return nil, err
-			}
-			return &clientCert, nil
-		}
-	}
+	var getClientCertificate = getClientCertFn(a.cfg)
```

Comment:
The change looks sound semantically, but the reason someone would ever make a function is to re-use it. Would this be re-used somewhere?

Reply:
I donot think this will be reused at least from foreseen. Abstract functions has another meaning to make it easy to maintain.